src/services/resource.js

```diff
@@ -206,4 +216,219 @@ module.exports = class resourceHelper {
 			throw error
 		}
 	}
+
+	static async upForReview(user_id, organization_id, roles, queryParams, searchText = '', page, limit) {
+		try {
+			let result = {
+				data: [],
+				count: 0,
+			}
+			let sort = {
+				sort_by: common.CREATED_AT,
+				order: common.SORT_DESC,
+			}
+			let finalResourceIds = []
+			let resourceIdsToBeRemoved = []
+
+			// fetch orgnization based configurations for each resources
+			const configList = await configs.list(organization_id)
+
+			// org based resource review type look up
+			const resourceWiseReviewType = configList.result.reduce((acc, item) => {
+				acc[item.resource_type] = item.review_type
+				return acc
+			}, {})
+
+			// fetch all resource type set as sequential review in the organization
+			const resourceTypesInSequentialReview = Object.keys(resourceWiseReviewType).filter(
+				(key) => resourceWiseReviewType[key] === common.REVIEW_TYPE_SEQUENTIAL
+			)
+
+			// fetch all resource type set as parallel review in the organization
+			const resourceTypesInParallelReview = Object.keys(resourceWiseReviewType).filter(
+				(key) => resourceWiseReviewType[key] === common.REVIEW_TYPE_PARALLEL
+			)
+
+			const userRoleTitles = [...new Set(roles.map((item) => item.title))]
+			// fetch review level of the reviewer
+			const reviewLevelDetails = await reviewStagesQueries.findAll(
+				{
+					organization_id,
+					role: {
+						[Op.in]: userRoleTitles,
+					},
+					resource_type: {
+						[Op.in]: Object.keys(resourceWiseReviewType),
+					},
+				},
+				{ attributes: ['resource_type', 'level'], order: [['level', 'ASC']] }
+			)
+			const resourceWiseLevels = {} //resource wise levels look up for the given reviewer
```

Comment:
@adithyadinesh0412 this can be optimzed
 const resourceWiseLevels = reviewLevelDetails.reduce((acc, item) => {

            acc[item.resource_type] = item.level;

            return acc;

        }, {});